editor/script_create_dialog.cpp

```diff
@@ -82,26 +82,20 @@ bool ScriptCreateDialog::_validate(const String &p_string) {
 	if (p_string.length() == 0)
 		return false;
 
-	String path_chars = "\"res://";
-	bool is_val_path = ScriptServer::get_language(language_menu->get_selected())->can_inherit_from_file();
+	if (ScriptServer::get_language(language_menu->get_selected())->can_inherit_from_file() && p_string.begins_with("\"") && p_string.ends_with("\"")) {
```

Comment:
Using `p_string.is_quoted()` instead to also support paths enclosed in single quotes.